tests/integration/frameworks/dendrites/dendrite_integration_tests.py

```diff
@@ -0,0 +1,142 @@
+    "nupic.research.frameworks.dendrites.dendrite_cl_experiment"
+)
+
+vernon_mixins = pytest.importorskip("nupic.research.frameworks.vernon.mixins")
+cl_mixins = pytest.importorskip("nupic.research.frameworks.continual_learning.mixins")
+
+
+class LocalExperiment(
+    vernon_mixins.RezeroWeights,
+    cl_mixins.SynapticIntelligence,
+    dendrites_mixins.PrototypeContext,
+    dendrites_experiments.DendriteContinualLearningExperiment,
+):
+    # computes error loss with and without SI and make sure the
+    # SI regularization term is being used
+    def error_loss(self, output, target, reduction="mean"):
+
+        # computes dendrites loss without SI
+        dendrites_original_loss = super(
+            dendrites_experiments.DendriteContinualLearningExperiment, self
+        ).error_loss(output, target, reduction)
+
+        # computes dendrites loss with SI
+        dendrites_with_si_loss = super().error_loss(output, target, reduction)
+        diff = dendrites_with_si_loss - dendrites_original_loss
+
+        # Computes surrogate loss
+        regularization_term = torch.tensor(0.0, device=self.device)
+
+        if self.current_task > 1:
+            for name, param in self.named_si_parameters():
+
+                big_omega = self.big_omega[name]
+                old_param = self.stable_params[name]
+
+                regularization_term += (big_omega * ((param - old_param) ** 2)).sum()
+
+        regularization_term = self.c * regularization_term
+
+        assert diff == regularization_term
+
+        return dendrites_with_si_loss
+
+
+def fake_data(size=100, image_size=(1, 28, 28), train=False):
+    return FakeData(size=size, image_size=image_size, transform=ToTensor())
+
+
+# A relatively quick running experiment for debugging
+integration_test_config = dict(
+    experiment_class=LocalExperiment,
+    num_classes=10,
+    # Dataset
+    dataset_class=fake_data,
+    # Number of epochs
+    epochs=1,
+    batch_size=10,
+    # Model class. Must inherit from "torch.nn.Module"
+    model_class=dendrites.DendriticMLP,
+    model_args=dict(
+        input_size=784,
+        output_size=10,
+        hidden_sizes=[2000, 2000],  # Note we use 2000 hidden units instead of 2048 for
+        # a better comparison with SI and XdG
+        num_segments=10,
+        dim_context=784,
+        kw=True,
+        kw_percent_on=0.05,
+        dendrite_weight_sparsity=0.0,
+        weight_sparsity=0.5,
+        context_percent_on=0.1,
+    ),
+    # Optimizer class class arguments passed to the constructor
+    optimizer_args=dict(lr=0.1),
+    fast_params=[".*"],  # <- all params get updated in inner loop
+    # Suppress logging.
+    log_level="NOTSET",
+    si_args=dict(
+        c=0.1,
+        damping=0.1,
+    ),
+)
+
+
+class DendriteIntegrationTest(unittest.TestCase):
+    """
+    This is a test class for testing Dendrite integration
+    with Vernon and Continual Learning
+    """
+
+    def test_base_experiment(self):
```

Comment:
This test is specific to "SynapticIntelligence" mixin, the test name should reflect that. 
Maybe `test_si_mixin` or something like that